modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -155,7 +158,22 @@ public synchronized List<ServiceRegistration<?>> deploy(
 					objectDefinition, _objectDefinitionLocalService,
 					_objectEntryManagerRegistry.getObjectEntryManager(
 						objectDefinition.getStorageType()),
-					_objectFieldLocalService, _objectRelationshipLocalService,
+					_objectFieldLocalService,
+					ListUtil.filter(
+						_objectFields,
```

Comment:
@pedro-oliveira446 did you get the chance to check what happens when we add a new custom object (using the UI, for example)?! How this lifecycle works? Will these variables be invalidated?

Check:
1 - Create a new custom object
2 - Create some fields for this object definition
3 - Put a breakpoint here
4 - Publish the object

The _objectFields must be null, right?!